teuthology/task/internal/__init__.py

```diff
@@ -314,8 +314,12 @@ def fetch_binaries_for_coredumps(path, remote):
             remote_path = remote.sh(['which', dump_program]).rstrip()
 
             # Pull remote program into coredump folder:
-            remote._sftp_get_file(remote_path, os.path.join(coredump_path,
-                                                            dump_program))
+            local_path = os.path.join(coredump_path,
+                                      dump_program.lstrip(os.path.sep))
+            local_dir = os.path.dirname(local_path)
+            if local_dir and not os.path.exists(local_dir):
```

Comment:
i don't think `local_dir` can be empty here. and there is no need to check if `local_dir` exists. one can simply put
```py
os.makedirs(local_dir, exists_ok=True)
```

Reply:
> i don't think `local_dir` can be empty here. and there is no need to check if `local_dir` exists. one can simply put
> 
> ```python
> os.makedirs(local_dir, exists_ok=True)
> ```

it can, be very easy, as in example see commit's description, for case when it is `/usr/bin/python3`
there is no `path/to/job/archive/remote/usr/bin` is never created.

Reply:
@kshtsk by "empty" i mean the string is empty. i think i did read the commit message.

Reply:
```exists_ok=True``` is not supported in py2

Reply:
ah, you mean it will always be at least 'coredump' if the code above will never changed

Reply:
i mean it always starts with `coredump_path`. and i am not going to argue with you if "the code above will never changed" or not.

Reply:
so you want to have this:
```
if not os.path.exists(local_dir):
    . . .
```